sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/KinesisIO.java

```diff
@@ -201,11 +222,22 @@ public Read withMaxReadTime(Duration maxReadTime) {
       return toBuilder().setMaxReadTime(maxReadTime).build();
     }
 
+    /**
+     * Specifies the maximum acceptable record lateness.
+     * When this limit is exceeded the runner might decide to scale the amount of resources
+     * allocated to the pipeline in order to speed up ingestion.
+     */
+    public Read withAllowedRecordLateness(Duration allowedRecordLateness) {
```

Comment:
As I wrote in general comment, this parameter should disable scaling up when events are late not more than allowedRecordLateness. So maybe I will rename it to "upToDateThreshold" to not confuse it with AllowedLateness concept in Beam. What do you think?

You are right that current implementation will actually never check the backlog size due to the way that reader's watermark is implemented. I'm planning to fix this in https://issues.apache.org/jira/browse/BEAM-2467. Maybe I should start the work with that other ticket but I noticed the dependency after submitting this PR. I've also reported 2 other improvements for this Kinesis source that will be submitted once these 2 are ready.